sdk/core/core-http/lib/serviceClient.ts

```diff
@@ -153,7 +151,7 @@ export class ServiceClient {
     if (Array.isArray(options.requestPolicyFactories)) {
       requestPolicyFactories = options.requestPolicyFactories;
     } else {
-      let credentialsOrFactory: ServiceClientCredentials | RequestPolicyFactory | undefined = undefined;
+      let credentialsOrFactory: RequestPolicyFactory | undefined = undefined;
       if (isTokenCredential(credentials)) {
```

Comment:
`isTokenCredential` is now going to return `true` for the credentials from the new 3.0.0 of ms-rest-nodeauth
In that case, arent we unnecessarily creating the wrapper RequestPolicyFactory here?

Reply:
The wrapper is used to make sure the `baseUri` gets propagated to the `scope` parameter of `BearerTokenAuthenticationPolicy` at the time of first use.  It's true that `scope` is effectively ignored `ms-rest-nodeauth` credentials since they are configured differently, but I don't know that it's worth distinguishing between a cred that is from `ms-rest-nodeauth` or not when creating the wrapper.  It'd just add extra complexity to something that's already a hack.

Reply:
Sorry for dragging this, but the credentials that is passed to the constructor is only `TokenCredential` at this point. So, I would expect that we need to wrap the credentials using this wrapper at all times. I don't see us falling into the `else` block below
```javascript
if (isTokenCredential(credentials)) {
 // wrap
} else {
  // use as is 
}
```

Unless, we are expecting our javascript users to pass in something that is either the old `ServiceClientCredentials` or some other type which is a `RequestPolicyFactory`

If the old `ServiceClientCredentials` is passed, then it gets ignored as we don't support signing policy any more.

Might I suggest the below:
```javascript
if (isTokenCredential(credentials)) {
 // wrap
} else if (isRequestPolicyFactory(credentials)) {
  // use as is 
}
```
And then ignore https://github.com/Azure/azure-sdk-for-js/pull/4367/files#r306129823

Reply:
I've cleaned up the logic here and in `createDefaultRequestPolicyFactories`, I think it's clearer now.  Thanks for the feedback!